### ISSUE TITLE
sti hangs in docker pull if tag is invalid

### ISSUE THREAD

**Issue by bparees** [openshift/source-to-image#179]
run STI and provide no tag on the image name, for an image repo where there is no "latest" tag, result is STI hangs forever in "pulling image"... seems like it must not be handling an error correctly:

$ docker pull ce-registry.usersys.redhat.com/jboss-webserver3/tomcat8-openshift
FATA[0005] Tag latest not found in repository ce-registry.usersys.redhat.com/jboss-webserver3/tomcat8-openshift

$ sti build https://github.com/bparees/session-app ce-registry.usersys.redhat.com/jboss-webserver3/tomcat8-openshift badout --loglevel=5
I0506 18:54:38.631805 07628 docker.go:173] Pulling image ce-registry.usersys.redhat.com/jboss-webserver3/tomcat8-openshift

(never errors/returns)

Note that if you provide an actual invalid tag, it does immediately error out.  Not sure it's actually valid to have a repo with no "latest" tag in it?

$ sti build https://github.com/bparees/session-app ce-registry.usersys.redhat.com/jboss-webserver3/tomcat8-openshift:badtag badout --loglevel=5
I0506 18:55:08.986159 08143 docker.go:173] Pulling image ce-registry.usersys.redhat.com/jboss-webserver3/tomcat8-openshift:badtag
I0506 18:55:12.403179 08143 docker.go:177] An error was received from the PullImage call: Tag badtag not found in repository ce-registry.usersys.redhat.com/jboss-webserver3/tomcat8-openshift

**Comment by bparees** [comment]
@soltysh fyi.

**Comment by soltysh** [comment]
@bparees addressed in #202, please close.

**Comment by bparees** [comment]
@soltysh your changes don't address the issue I reported which was the case where :latest was specified on the ref, but there was no :latest tag in the registry.  your changes just add a :latest tag if no tag is specified.

that said, this seems to be working now (as in it immediately returns an error due to the invalid tag), so maybe docker fixed their registry:

$  sti build https://github.com/bparees/session-app ce-registry.usersys.redhat.com/jboss-webserver3/tomcat8-openshift badout --loglevel=5
I0527 20:24:16.810998 30901 util.go:34] Using bparees@redhat.com credentials for pulling ce-registry.usersys.redhat.com/jboss-webserver3/tomcat8-openshift

Builder Image:		ce-registry.usersys.redhat.com/jboss-webserver3/tomcat8-openshift
Source:			https://github.com/bparees/session-app
Output Image Tag:	badout
Incremental Build:	disabled
Remove Old Build:	disabled
Force Pull:		enabled
Quiet:			disabled
Layered Build:		disabled
Docker Endpoint:	unix:///var/run/docker.sock
Docker Pull Config:	/home/bparees/.dockercfg
Docker Pull User:	bparees

I0527 20:24:16.812186 30901 docker.go:187] Pulling image ce-registry.usersys.redhat.com/jboss-webserver3/tomcat8-openshift:latest
I0527 20:24:21.098681 30901 docker.go:190] An error was received from the PullImage call: Tag latest not found in repository ce-registry.usersys.redhat.com/jboss-webserver3/tomcat8-openshift
E0527 20:24:21.098725 30901 main.go:246] An error occurred: unable to get ce-registry.usersys.redhat.com/jboss-webserver3/tomcat8-openshift:latest
E0527 20:24:21.098740 30901 main.go:247] Suggested solution: check image name, or if using local image add --force-pull=false flag
I0527 20:24:21.098749 30901 main.go:249] Details: Tag latest not found in repository ce-registry.usersys.redhat.com/jboss-webserver3/tomcat8-openshift
E0527 20:24:21.098758 30901 main.go:253] If the problem persists consult the docs at https://github.com/openshift/source-to-image/tree/master/docs.Eventually reach us on freenode #openshift or file an issue at https://github.com/openshift/source-to-image/issues providing us with a log from your build using --loglevel=3

**Comment by soltysh** [comment]
>that said, this seems to be working now (as in it immediately returns an error due to the invalid tag), so maybe docker fixed their registry:

@bparees that's exactly what I saw when testing this. Still the message returned from the server was not clear, from top of my head it was something about image metadata, which isn't clear IMHO. I've then decided to add that :latest tag to builder images without it specified. At least this way the message is clear that there's no such tag as 'latest' on image xyz. Besides that should also address those registries that were not updated.
